library/common.h

```diff
@@ -236,15 +244,22 @@ static inline void mbedtls_xor_no_simd(unsigned char *r,
         uint64_t x = mbedtls_get_unaligned_uint64(a + i) ^ mbedtls_get_unaligned_uint64(b + i);
         mbedtls_put_unaligned_uint64(r + i, x);
     }
+    // This if statement helps some compilers (e.g., IAR) optimise out the byte-by-byte tail case
```

Comment:
@mschulz-at-hilscher This gets IAR codegen for this function on a par with recent clang; maybe this also does the job for old versions of gcc?

Reply:
I will give it a try.